components/service/glean/src/test/java/mozilla/components/service/glean/scheduler/MetricsPingSchedulerTest.kt

```diff
@@ -440,6 +444,34 @@ class MetricsPingSchedulerTest {
         assertTrue(getWorkerStatus(MetricsPingWorker.TAG).isEnqueued)
     }
 
+    @Test
+    fun `startupCheck happens when returning from background when Glean is already initialized`() {
+        // For faking calling the lifecycle observer.
+        val lifecycleRegistry = LifecycleRegistry(mock(LifecycleOwner::class.java))
+        val gleanLifecycleObserver = GleanLifecycleObserver()
+        lifecycleRegistry.addObserver(gleanLifecycleObserver)
+
+        // Set the last sent date to yesterday.
+        val mpsSpy = spy(MetricsPingScheduler(ApplicationProvider.getApplicationContext()))
+
+        // Simulate the first foreground event after the application starts.
+        lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_START)
+
+        // Initialize Glean
+        Glean.initialize(getContextWithMockedInfo())
+        Glean.metricsPingScheduler = mpsSpy
+
+        // Simulate going to background.
+        lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_STOP)
+
+        // Simulate returning to the foreground.
+        lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_START)
+
+        // Verify that we're performing the startup check. Using 'timeout' to allow time for
+        // the lifecycle event to process.
+        verify(mpsSpy, timeout(1000).times(1)).startupCheck()
```

Comment:
How was this timeout picked? Let's use an higher one, maybe 5 s is enough to account for slow CI, we had bugs before about that and had [to bump to 5s](https://github.com/mozilla-mobile/android-components/blob/dcf534f57024dfae161e4d47541e590db23a8c7b/components/service/glean/src/test/java/mozilla/components/service/glean/TestUtil.kt#L185)

Reply:
It could actually be much less than this. I think 100ms worked pretty consistently so I figured 10x that for CI, but this is a cool function and bails out early if it can verify before the timeout (at least according to the docs) so I'll go ahead and up this as you request.